src/Library/demos/View Switcher/main.blp

```diff
@@ -0,0 +1,150 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.Window {
+  modal: true;
+  width-request: 360;
+  height-request: 360;
+  content:
+  Box {
+    orientation: vertical;
+
+    Adw.HeaderBar {
+      centering-policy: strict;
+      title-widget:
+      Adw.ViewSwitcherTitle switcher_title {
+        stack: stack;
+        title: _("AdwViewSwitcher Demo");
+      };
+    }
+
+    Adw.ViewStack stack {
+      vexpand: true;
+
+      Adw.ViewStackPage page1 {
+        name: "page1";
+        title: _("Favorites");
+        icon-name: "star-filled-rounded-symbolic";
+        use-underline: true;
+        child:
+        Box{
+          orientation: vertical;
+          valign: center;
+
+          Image {
+            margin-top: 30;
+            icon-name: bind page1.icon-name;
+            pixel-size: 128;
+          }
+
+          Label {
+            margin-top: 24;
+            margin-bottom: 24;
+            margin-start: 24;
+            margin-end: 24;
+            label: bind page1.title;
+          }
+
+          LinkButton {
+            label: "API Reference";
+            uri: "https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1.2/class.ViewSwitcher.html";
+          }
+
+          LinkButton {
+            label: "Human Interface Guidelines";
+            uri: "https://developer.gnome.org/hig/patterns/nav/view-switchers.html";
+          }
+        };
+      }
+
+      Adw.ViewStackPage page2 {
+        name: "page2";
+        title: _("Recents");
+        icon-name: "clock-alt-symbolic";
+        use-underline: true;
+        child:
+        Box{
+         orientation: vertical;
+         valign: center;
+
+          Image {
+            margin-top: 30;
+            icon-name: bind page2.icon-name;
+            pixel-size: 128;
+          }
+
+          Label {
+            margin-top: 24;
+            margin-bottom: 24;
+            margin-start: 24;
+            margin-end: 24;
+            label: bind page2.title;
+          }
+        };
+      }
+
+      Adw.ViewStackPage page3 {
+        name: "page3";
+        title: _("Notifications");
+        icon-name: "bell-symbolic";
+        badge-number: 3;
+        needs-attention: true;
+        use-underline: true;
+        child:
+        Box{
```

Comment:
This demo would be much more helpful with a bit of code showing how to control the `Adw.ViewSwitcher` appearance by modifying the `needs-attention` and `badge-number` properties.

Something like a `Gtk.ListBox` with a number of rows matching the `badge-number`, which is updates as rows are removed (i.e. notifications marked as read).